### PR TITLE
BF+RF: Consistent and more "thorough" handling of fake HOMEs (including Windows) (master)

### PR DESCRIPTION
**Pull request by yarikoptic**
https://github.com/datalad/datalad/pull/5345 (and current maint, with just spotted typo introduced in 193bd76b19beaf02fcdef984febdec58f1cbcb38, heh heh, will fixup in maint) with merge of master. Let's see if appveyor is happy here

### CODE REVIEW

**Comment by yarikoptic** [comment]
ha -- tests pass here (master) but not for (maint) on windows.  Here though coverage report showing that [windows-specific block](https://github.com/datalad/datalad/pull/5348/files#) not being covered suggesting that new appveyor setup does not submit coverage "properly" at all.  I did spend time fixing it for old appveyor setup just recently (https://github.com/datalad/datalad/pull/5223) so it is still sad (even if [not news](https://github.com/datalad/datalad/pull/5330#issuecomment-760241674))

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -299,9 +299,26 @@ Alternatively, or complimentary to that, you can use `tox` -- there is a `tox.in
 file which sets up a few virtual environments for testing locally, which you can
 later reuse like any other regular virtualenv for troubleshooting.
 Additionally, [tools/testing/test_README_in_docker](tools/testing/test_README_in_docker) script can
-be used to establish a clean docker environment (based on any NeuroDebian-supported
+be used to establish a clean docker environment (based on any NtesteuroDebian-supported
```
Comment (mih) [inline CONTRIBUTING.md:302]: Unless this is the new bitcoin wallet for NeuroDebian, I guess it should be removed again ;-)

**Review by mih — APPROVED** [review]
LGTM! Thanks for investigating this!

Intuitively, I would have placed `get_home_envvars()` into `datalad.tests.utils`, but I am OK with your placement too.

The incomprehensible diff in `CONTRIBUTING` should be fixed, but otherwise this is cool.

File: CONTRIBUTING.md (hunk 1 above)
Comment (yarikoptic) [reply CONTRIBUTING.md:302]: Typo was fixed in maint already... I will then merge maint and this into master locally and push

**Pull request merged**